docs/syntax_grammar.md

```diff
@@ -104,7 +104,7 @@ not allowed.
     for_stmt   = "for" range NL
                     statements
                  "end" NL .
-    range      = ident ( ":=" | "=" ) "range" range_args .
+    range      = [ ident ( ":=" | "=" ) ] "range" range_args .
```

Comment:
It seem like `=` is not allowed, this does not compile:
```
i:num
for i = range 3 
    print i
end
```

Reply:
ok. I have removed that from the grammar, and (you're not going to like this) added a TODO to allow assignment in for loops. :moneybag: